From: Ahmad Fatoum <a.fatoum@pengutronix.de>
Date: Sat, 21 May 2022 08:06:21 +0200
Subject: [PATCH] ARM: lxatac: add FIP barebox update handler

We boot from eMMC and with recent TF-A rework both SSBL (as part of FIP)
and FSBL can be located in the eMMC boot partition. Use
stm32mp_bbu_mmc_fip_register to allow updating barebox from within
barebox atomically.

Signed-off-by: Ahmad Fatoum <a.fatoum@pengutronix.de>
---
 arch/arm/boards/lxa-tac/board.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/boards/lxa-tac/board.c b/arch/arm/boards/lxa-tac/board.c
index b616ec5e8348..8e7733e2ac00 100644
--- a/arch/arm/boards/lxa-tac/board.c
+++ b/arch/arm/boards/lxa-tac/board.c
@@ -11,6 +11,8 @@ static int tac_probe(struct device_d *dev)
 {
 	barebox_set_hostname("lxatac");
 
+	stm32mp_bbu_mmc_fip_register("mmc", "/dev/mmc1", BBU_HANDLER_FLAG_DEFAULT);
+
 	return 0;
 }
 
